pyppeteer/frame_manager.py

```diff
@@ -149,60 +275,82 @@ def _onFrameNavigated(self, framePayload: dict) -> None:
 
         # Update frame payload.
         frame._navigated(framePayload)  # type: ignore
-        self.emit(FrameManager.Events.FrameNavigated, frame)
+        self.emit(Events.FrameManager.FrameNavigated, frame)
+
+    async def _ensureIsolatedWorld(self, name: str):
+        if name in self._isolatedWorlds:
+            return
+        self._isolatedWorlds.add(name)
+        await self._client.send(
+            'Page.addScriptToEvaluateOnNewDocument',
+            {
+                'source': f'//# sourceURL={EVALUATION_SCRIPT_URL}',
+                'worldName': name,
+            }
+        )
+        # todo wrap in debugerror?
+        await asyncio.gather(
+            *[self._client.send(
+                'Page.createIsolatedWorld',
+                {
+                    'frameId': frame._id,
+                    'grantUniversalAccess': True,
+                    'worldName': name,
+                }
+            ) for frame in self.frames()]
+        )
```

Comment:
Personally, I'm partial to non-early returns:
```suggestion
        if name not in self._isolatedWorlds:
            self._isolatedWorlds.add(name)
            await self._client.send(
                'Page.addScriptToEvaluateOnNewDocument',
                {
                    'source': f'//# sourceURL={EVALUATION_SCRIPT_URL}',
                    'worldName': name,
                }
            )
            # todo wrap in debugerror?
            await asyncio.gather(
                *[self._client.send(
                    'Page.createIsolatedWorld',
                    {
                        'frameId': frame._id,
                        'grantUniversalAccess': True,
                        'worldName': name,
                    }
                ) for frame in self.frames()]
            )
```
But really, it's not that big of a deal so this one is your call

Reply:
I somewhat prefer to save horizontal space here as with all of these long js names and string calls it gets polluted really fast :grimacing: